### PR TITLE
Gun locker now uses a radial menu

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

This PR replaces a gun locker HTML UI with a radial menu, which is used for taking various weaponry from it.

**BEFORE:**

GunLockerOLD

**AFTER:**

GunLockerNEW

## Why It's Good For The Game

Better usability.

## Changelog
:cl: Arkatos
add: Gun locker now uses a radial menu.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/guncase.dm [hunk 1]
```diff
@@ -60,36 +60,58 @@
...
-			update_icon()
+	var/pick = show_radial_menu(user, src, items, custom_check = CALLBACK(src, .proc/check_menu, user), radius = 36, require_near = TRUE)
+	if(!pick)
+		return
+
+	var/weapon_reference = display_names[pick]
+	var/obj/item/O = locate(weapon_reference) in contents
+	if(!O || !istype(O))
+		return
+	if(ishuman(user))
```
Comment (ShizCalev) [inline code/game/objects/structures/guncase.dm:96]: early return

**Comment by davethwave** [comment]
We have such magical things? Why doesn't the armory use them? This would be perfect for armories.

File: code/game/objects/structures/guncase.dm (hunk 1 above)
Comment (Arkatos1) [reply code/game/objects/structures/guncase.dm]: I put the check in the check_menu proc instead, with an early return. Fixed.

**Comment by nicbn** [comment]
You could add a little "x2" text to the option to symbolize there are 2 of this type of gun.

Not sure if there's already code for it tho, I'll add it later it there isn't

**Comment by Tlaltecuhtli** [comment]
> We have such magical things? Why doesn't the armory use them? This would be perfect for armories.

i only saw them during telesci era, they existed to prevent guns from vanishing in 3 min

**Comment by Dennok** [comment]
> > We have such magical things? Why doesn't the armory use them? This would be perfect for armories.
> 
> i only saw them during telesci era, they existed to prevent guns from vanishing in 3 min

Why no one invented tele jammer that prevent to teleport on certain places and take power, so it can be bypassed by power outage.

File: code/game/objects/structures/guncase.dm [hunk 2]
```diff
@@ -60,36 +60,57 @@
...
-	popup.open(FALSE)
+	var/list/display_names = list()
+	var/list/items = list()
+	for(var/i in 1 to contents.len)
+		var/obj/item/I = contents[i]
+		display_names[I.name + " ([i])"] = REF(I)
+		var/image/item_image = image(icon = I.icon, icon_state = I.icon_state)
+		if(length(I.overlays))
+			item_image.copy_overlays(I)
+		items += list(I.name + " ([i])" = item_image)
```
Comment (Rohesie) [inline code/game/objects/structures/guncase.dm:86]:
```suggestion
	for(var/i in 1 to length(contents))
		var/obj/item/thing = contents[i]
		display_names[thing.name + " ([thing])"] = REF(thing)
		var/image/item_image = image(icon = thing.icon, icon_state = thing.icon_state)
		if(length(thing.overlays))
			item_image.copy_overlays(thing)
		items += list(thing.name + " ([thing])" = item_image)
```
Avoid single upper-case letters for variables, please.

File: code/game/objects/structures/guncase.dm [hunk 3]
```diff
@@ -60,36 +60,57 @@
...
+	var/pick = show_radial_menu(user, src, items, custom_check = CALLBACK(src, .proc/check_menu, user), radius = 36, require_near = TRUE)
+	if(!pick)
+		return
+
+	var/weapon_reference = display_names[pick]
+	var/obj/item/O = locate(weapon_reference) in contents
+	if(!O || !istype(O))
+		return
+	if(!user.put_in_hands(O))
+		O.forceMove(get_turf(src))
```
Comment (Rohesie) [inline code/game/objects/structures/guncase.dm:97]:
```suggestion
	var/obj/item/weapon = locate(weapon_reference) in contents
	if(!istype(weapon))
		return
	if(!user.put_in_hands(weapon))
		weapon.forceMove(get_turf(src))
```

File: code/game/objects/structures/guncase.dm (hunk 2 above)
Comment (Arkatos1) [reply code/game/objects/structures/guncase.dm]: Fixed.

File: code/game/objects/structures/guncase.dm (hunk 3 above)
Comment (Arkatos1) [reply code/game/objects/structures/guncase.dm]: Fixed.

File: code/game/objects/structures/guncase.dm [hunk 4]
```diff
@@ -77,13 +77,13 @@
...
-	for(var/i in 1 to contents.len)
-		var/obj/item/I = contents[i]
-		display_names[I.name + " ([i])"] = REF(I)
-		var/image/item_image = image(icon = I.icon, icon_state = I.icon_state)
-		if(length(I.overlays))
-			item_image.copy_overlays(I)
-		items += list(I.name + " ([i])" = item_image)
+	for(var/i in 1 to length(contents))
+		var/obj/item/thing = contents[i]
+		display_names[thing.name + " ([i])"] = REF(thing)
```
Comment (Rohesie) [inline code/game/objects/structures/guncase.dm:82]:
```suggestion
		display_names["[thing.name] ([i])"] = REF(thing)
```
This is actually much faster.

File: code/game/objects/structures/guncase.dm [hunk 5]
```diff
@@ -77,13 +77,13 @@
...
+		var/image/item_image = image(icon = thing.icon, icon_state = thing.icon_state)
+		if(length(thing.overlays))
+			item_image.copy_overlays(thing)
+		items += list(thing.name + " ([i])" = item_image)
```
Comment (Rohesie) [inline code/game/objects/structures/guncase.dm:86]:
```suggestion
		items += list("[thing.name] ([i])" = item_image)
```

File: code/game/objects/structures/guncase.dm (hunk 4 above)
Comment (Arkatos1) [reply code/game/objects/structures/guncase.dm]: Fixed.

File: code/game/objects/structures/guncase.dm (hunk 5 above)
Comment (Arkatos1) [reply code/game/objects/structures/guncase.dm]: Fixed.

**Comment by davethwave** [comment]
Once this gets merged I'll try to get this re-added for standard armory use if I can figure out map making stuff.

**Pull request merged**